pyccel/parser/semantic.py

```diff
@@ -1933,6 +1948,8 @@ def _visit(self, expr):
         pyccel.ast.basic.PyccelAstNode
             AST object which is the semantic equivalent of expr.
         """
+        if getattr(expr, 'pyccel_staging', 'syntactic') == 'semantic':
+            return expr
```

Comment:
Why using `getattr` with a default value if `expr` is of type `PyccelAstNode`, which has the `pyccel_staging` property?

Reply:
Because sometimes `expr` is a `PyccelSymbol` which isn't a `PyccelAstNode`

Reply:
The docstring says otherwise...

Reply:
Fixed 18c473a6f54633e9e3d52ec60e5a576fee28ca2d